PR: https://github.com/odoo/odoo/pull/

From: 366aa2907e792b3cc30cb6f4300ddad6146118fe
From: dht-odoo
Date: 2021-06-07 05:21:40

Structural Changes: 8
Total Changes: 8

[IMP] stock: improves field type from text to html

Replace text fields to html fields as we have our own 'OdooEditor'.
Indeed, it gives more options to users in the way they format their
content without weighting too much on the UI
(tools appear on demand and not by default).

Models -> Fieldss

1) stock.location -> comment
2) stock.picking -> note

Task Id: 2499504

X-original-commit: 96f5316a219568add44bee528245c2c8e49c3bbc

================================= pseudo patch: =================================

--- a/addons/stock/models/product.py
+++ b/addons/stock/models/product.py
@@ -626,8 +626,8 @@ class ProductTemplate(models.Model):
         ('lot', 'By Lots'),
         ('none', 'No Tracking')], string="Tracking", help="Ensure the traceability of a storable product in your warehouse.", default='none', required=True)
     description_picking = fields.Text('Description on Picking', translate=True)
-    description_pickingout = fields.Text('Description on Delivery Orders', translate=True)
-    description_pickingin = fields.Text('Description on Receptions', translate=True)
+    description_pickingout = fields.Html('Description on Delivery Orders', translate=True)
+    description_pickingin = fields.Html('Description on Receptions', translate=True)
     qty_available = fields.Float(
         'Quantity On Hand', compute='_compute_quantities', search='_search_qty_available',
         compute_sudo=False, digits='Product Unit of Measure')

--- a/addons/stock/models/stock_location.py
+++ b/addons/stock/models/stock_location.py
@@ -51,7 +51,7 @@ class Location(models.Model):
         'stock.location', 'Parent Location', index=True, ondelete='cascade', check_company=True,
         help="The parent location that includes this location. Example : The 'Dispatch Zone' is the 'Gate 1' parent location.")
     child_ids = fields.One2many('stock.location', 'location_id', 'Contains')
-    comment = fields.Text('Additional Information')
+    comment = fields.Html('Additional Information')
     posx = fields.Integer('Corridor (X)', default=0, help="Optional localization details, for information purpose only")
     posy = fields.Integer('Shelves (Y)', default=0, help="Optional localization details, for information purpose only")
     posz = fields.Integer('Height (Z)', default=0, help="Optional localization details, for information purpose only")

--- a/addons/stock/models/stock_picking.py
+++ b/addons/stock/models/stock_picking.py
@@ -255,7 +255,7 @@ class Picking(models.Model):
         'Source Document', index=True,
         states={'done': [('readonly', True)], 'cancel': [('readonly', True)]},
         help="Reference of the document")
-    note = fields.Text('Notes')
+    note = fields.Html('Notes')
     backorder_id = fields.Many2one(
         'stock.picking', 'Back Order of',
         copy=False, index=True, readonly=True,
